include/vcpkg/base/expected.h

```diff
@@ -132,12 +132,24 @@ namespace vcpkg
         explicit constexpr operator bool() const noexcept { return !m_s.has_error(); }
         constexpr bool has_value() const noexcept { return !m_s.has_error(); }
 
+        const T&& value_or_exit(const LineInfo& line_info) const&&
```

Comment:
Is this needed? Shouldn't this resolve to the `const&` overload?

Reply:
I didn't want to need to consider creating ambiguities here. If we're providing 3 it isn't that more complex to add 4 and avoid horrible overloading error messages.